src/database/dataLoading.js

```diff
@@ -36,3 +36,15 @@ export async function loadDataForFirstTime (db, dataSource) {
 
   await loadData(db, emojiData, dataSource, eTag)
 }
+
+export async function checkForUpdates (db, dataSource) {
+  try {
+    await doCheckForUpdates(db, dataSource)
+  } catch (err) {
+    // Checking for updates is not a critical operation, and it can fail if e.g. the picker is quickly removed
+    // and re-added to the DOM. We can safely ignore IndexedDB InvalidStateErrors here
```

Comment:
Why does rapid removal/addition cause an `InvalidStateError`? Should that error be handled closer to its source?

Reply:
IndexedDB is weird, man. I'd love to find a better solution.

Reply:
Another approach is to pass `AbortSignals`s down various method calls and abort when the picker is removed from the DOM, but I tried that out and it was a huge amount of added complexity.